src/components/Review/ReviewsList/ReviewsList.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import Button from '@/components/UI/Buttons/Button/Button'
+import ScrollUpBtn from '@/components/UI/Buttons/ScrollUpBtn/ScrollUpBtn'
+import React from 'react'
+import { apiDeleteProductReview } from '@/services/reviewService'
+import { useErrorHandler } from '@/services/apiError/apiError'
+import { useProductReviewsStore } from '@/store/reviewsStore'
+import { Review as ReviewType } from '@/types/ReviewType'
+import Loader from '@/components/UI/Loader/Loader'
+import Review from '@/components/Review/Review/Review'
+
+interface IReviewsList {
+  productId: string
+  reviews: ReviewType[]
+  hasNextPage: boolean
+  showMoreReviews: () => void
+  isFetchingNextPage: boolean | undefined
+  userReview: ReviewType | null
+}
+
+const ReviewsList: React.FC<IReviewsList> = ({
+  productId,
+  reviews,
+  hasNextPage,
+  showMoreReviews,
+  isFetchingNextPage,
+  userReview,
+}) => {
+  const {
+    setIsReviewFormVisible,
+    setIsReviewButtonVisible,
+    setIsRaitingFormVisible,
+    setShouldRevalidateStatistics,
+    setShouldRevalidateUserReview,
+    setShouldRevalidateReviews,
+  } = useProductReviewsStore()
+
+  const {
+    handleError,
+    // errorMessage
+  } = useErrorHandler()
+
+  const deleteReviewHandler = async (
+    productReviewId: string,
+  ): Promise<void> => {
+    try {
+      productReviewId &&
+        (await apiDeleteProductReview(productReviewId, productId))
+
+      setShouldRevalidateStatistics(true)
+      setShouldRevalidateReviews(true)
+      setShouldRevalidateUserReview(true)
+
+      setIsReviewFormVisible(false)
+      setIsReviewButtonVisible(true)
+      setIsRaitingFormVisible(false)
+    } catch (error) {
+      handleError(error)
+    }
+  }
+
+  const handleLikeComment = (productReviewId: string | null | undefined) => {
```

Comment:
these functions can be created like 
`const handleDislikeComment = (productReviewId: string | null | undefined)  => () => {`
then you don't to create handlers for each of them inside the `<Review>`